llvm/include/llvm/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddFunctions.h

```diff
@@ -0,0 +1,24 @@
+#ifndef LLVM_TRANSFORMS_PEEPHOLE_OBFUSCATE_I8_ADD_FUNCTION
+#define LLVM_TRANSFORMS_PEEPHOLE_OBFUSCATE_I8_ADD_FUNCTION
+
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Type.h"
+
+namespace llvm {
+
+static bool isBinaryAdd(Instruction const* instruction){
+	return instruction->getOpcode() == Instruction::Add &&
+           instruction->getNumOperands() == 2;
+}
+
+static bool isI8Instruction(Instruction const* instruction){
+	return instruction->getType()->isIntegerTy() && 
+	   instruction->getType()->getIntegerBitWidth() == 8;
+}
```

Comment:
Why did you decide to implement these function in a header?

Reply:
That seemed like a good solution in this case.